django/db/migrations/serializer.py

```diff
@@ -246,11 +247,14 @@ class RegexSerializer(BaseSerializer):
     def serialize(self):
         imports = {"import re"}
         regex_pattern, pattern_imports = serializer_factory(self.value.pattern).serialize()
-        regex_flags, flag_imports = serializer_factory(self.value.flags).serialize()
+        # We turn off default implicit flags, because regexes with the same
+        # implicit and explicit flags are not equal.
+        flags = self.value.flags ^ re.compile('').flags
```

Comment:
Without `xor` default flags (i.e. `re.UNICODE`) are always set explicit after serialization. I think that is something that we want to avoid.

Before this PR:
```python
>>> serializer_factory(re.compile(r'^\w+$')).serialize()
("re.compile('^\\\\w+$', 32)", {'import re'})
>>> serializer_factory(re.compile(r'^\w+$', 32)).serialize()
("re.compile('^\\\\w+$', 32)", {'import re'})
```
After this PR:
```python
>>> serializer_factory(re.compile(r'^\w+$')).serialize()
("re.compile('^\\\\w+$')", {'import re'})
>>> serializer_factory(re.compile(r'^\w+$', 32)).serialize()
("re.compile('^\\\\w+$')", {'import re'})
```